priam/src/main/java/com/netflix/priam/backup/SnapshotBackup.java

```diff
@@ -180,11 +190,50 @@ private void notifyObservers() {
     protected void processColumnFamily(String keyspace, String columnFamily, File backupDir) throws Exception {
 
         File snapshotDir = getValidSnapshot(backupDir, snapshotName);
-        // Add files to this dir
-        if (null != snapshotDir)
-            abstractBackupPaths.addAll(upload(snapshotDir, BackupFileType.SNAP));
-        else
+
+        if (snapshotDir == null) {
             logger.warn("{} folder does not contain {} snapshots", backupDir, snapshotName);
+            return;
+        }
+
+        findForgottenFiles(snapshotDir);
+        // Add files to this dir
+        abstractBackupPaths.addAll(upload(snapshotDir, BackupFileType.SNAP));
+    }
+
+    private void findForgottenFiles(File snapshotDir) {
+        try {
+            Collection<File> snapshotFiles = FileUtils.listFiles(snapshotDir, FileFilterUtils.fileFileFilter(), null);
+            File columnfamilyDir = snapshotDir.getParentFile().getParentFile();
+            Collection<File> columnfamilyFiles = FileUtils.listFiles(columnfamilyDir, FileFilterUtils.fileFileFilter(), null);
+
+            for (File file : snapshotFiles) {
+                //Get its parent directory file based on this file.
+                File originalFile = new File(columnfamilyDir, file.getName());
+                columnfamilyFiles.remove(originalFile);
+            }
+
+            if (columnfamilyFiles.size() == 0)
+                return;
+
+            //There might be new SSTables since we took snapshot. Remove them from this list.
```

Comment:
I think you need to document more of the algorithm here. From what I can tell you're considering any file which is older than the snapshot instant to be forgotten. I don't think this is what you want. You probably want files which are older than gc_grace, not in the snapshot, and not -tmp- files.

I think the algorithm needs to

Reply:
Why not? If a file which was written before the snapshot time is not part of the snapshot, isn't that forgotten (or tmp file)? I don't understand the need for GC_GRACE from your comments. 
Anything which is -tmp- 👍 : Yes I forgot about it as there could have a compaction in progress while snapshot was executed. We should remove any -tmp- file from CF for comparison purposes. Also, in reality, update time should be updated for -tmp- files if they are actively being written to but race condition while code execution (after snapshot) and compaction flush make me believe this should be done.

Reply:
Cassandra doesn't instantly delete sstables, especially as compaction or repair byproducts there can be sstables that are not in the view but maybe a few hours old. This is fine as long as the files are not older than gc_grace (in which case they could contain deleted data).